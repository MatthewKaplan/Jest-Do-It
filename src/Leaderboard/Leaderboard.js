import React, {Component} from 'react';
import '../Styles/_Leaderboard.scss';

export class Leaderboard extends Component {
  constructor() {
    super();
    this.state = {
      list: [] 
    }
  }

  componentDidMount() {
    let sortedLoaderboardArr = this.props.leaderboardArr.map(player => player).sort((a,b) => b.score - a.score);
    this.setState({
      list: sortedLoaderboardArr
    })
  }

  render() {
    let userlist = this.state.list.map((user, i) => <User name={  user.name } rank={ i + 1 } answeredCorrectly={ user.score } />);
    return (
      <div className="container">
        <div className="leadheader">
          <h2>Leaderboard</h2>
        </div>
        <div className="row colheader">
        <div className="col-xs-1">
          <h4>#</h4>
        </div>
        <div className="col-xs-5">
          <h4>Name</h4>
        </div>
        <div className="col-xs-3 recent">
          <h4>Answered Correctly</h4>
        </div>
      </div>
        { userlist }
      </div>
    )
  }
}

const User = ({ rank, answeredCorrectly, name }) => {
  return (
    <div className="row users  vcenter">
      <div className="col-xs-1 rank">
        <h4>{ rank }</h4>
      </div>
      <div className="col-xs-5 name">
        <h4>{ name }</h4>
      </div>
      <div className="col-xs-3">
        <h4>{ answeredCorrectly }</h4>
      </div>
    </div>
  )
}

export default Leaderboard;