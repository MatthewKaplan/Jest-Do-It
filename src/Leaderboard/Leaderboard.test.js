import React from 'react';
import { Leaderboard } from './Leaderboard';
import { shallow } from 'enzyme';
