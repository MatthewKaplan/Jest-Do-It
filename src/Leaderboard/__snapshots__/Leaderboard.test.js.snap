// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard should match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="leadheader"
  >
    <h2>
      Leaderboard
    </h2>
  </div>
  <div
    className="row colheader"
  >
    <div
      className="col-xs-1"
    >
      <h4>
        #
      </h4>
    </div>
    <div
      className="col-xs-5"
    >
      <h4>
        Name
      </h4>
    </div>
    <div
      className="col-xs-3 recent"
    >
      <h4>
        Answered Correctly
      </h4>
    </div>
  </div>
  <User
    answeredCorrectly={2}
    name="Kevin"
    rank={1}
  />
  <User
    answeredCorrectly={1}
    name="Matthew"
    rank={2}
  />
</div>
`;
